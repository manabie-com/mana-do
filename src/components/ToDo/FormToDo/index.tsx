import React, { useRef } from "react";
import { useHistory } from "react-router-dom";
import { Input } from "components/commons";

interface Props {
  onCreateToDo: (value: string) => Promise<boolean>;
}

const FormToDo = (props: Props) => {
  const { onCreateToDo } = props;
  const inputRef = useRef<HTMLInputElement>(null);
  const history = useHistory();

  const _handleKeyDown = async (e: React.KeyboardEvent<HTMLInputElement>) => {
    if (e.key === "Enter" && inputRef.current) {
      try {
        await onCreateToDo(inputRef.current.value);
        inputRef.current.value = "";
      } catch (e) {
        if (e.response.status === 401) {
          history.push("/");
        }
      }
    }
  };
  return (
    <div className="Todo__creation">
      <Input
        ref={inputRef}
        className="Todo__input"
        placeholder="What need to be done?"
        onKeyDown={_handleKeyDown}
      />
    </div>
  );
};

export default FormToDo;
