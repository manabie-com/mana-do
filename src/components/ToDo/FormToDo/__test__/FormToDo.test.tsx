import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup, fireEvent } from "@testing-library/react";

import FormToDo from "components/ToDo/FormToDo";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";

const propsTest = {
  onCreateToDo: jest.fn(),
};

let container: HTMLElement;

beforeEach(() => {
  container = render(<FormToDo {...propsTest} />).container;
});

afterEach(cleanup);

test("FormToDo: render without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<FormToDo {...propsTest} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test("FormToDo: test event keydown", () => {
  const usernameEl = container.querySelector("input") as HTMLInputElement;
  fireEvent.change(usernameEl, { target: { value: "new task 1" } });
  fireEvent.keyDown(usernameEl, { key: "Enter", code: "Enter", charCode: 13 });
  expect(propsTest.onCreateToDo).toBeCalled();
});

test("SignInPage: match snapshot", () => {
  const tree = renderer.create(<FormToDo {...propsTest} />).toJSON();
  expect(tree).toMatchSnapshot();
});
