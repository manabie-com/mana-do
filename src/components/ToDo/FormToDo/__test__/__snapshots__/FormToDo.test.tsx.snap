// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInPage: match snapshot 1`] = `
<div
  className="Todo__creation"
>
  <div
    className="root-input Todo__input"
  >
    <input
      className="input-custom input-size-md"
      onKeyDown={[Function]}
      placeholder="What need to be done?"
    />
  </div>
</div>
`;
